src/grt/src/fastroute/include/FastRoute.h

```diff
@@ -120,28 +125,344 @@ class FastRouteCore
+  void routeSegLFirstTime(Segment* seg);
+  void spiralRoute(int netID, int edgeID);
+  void routeLVEnew(int netID, int edgeID, int threshold, int enlarge);
+
+  // ripup functions
+  void ripupSegL(Segment* seg);
+  void ripupSegZ(Segment* seg);
+  void newRipup(TreeEdge* treeedge,
+                       TreeNode* treenodes,
+                       int x1,
+                       int y1,
+                       int x2,
+                       int y2,
+                       int netID);
+  bool newRipupCheck(TreeEdge* treeedge,
+                            int x1,
+                            int y1,
+                            int x2,
+                            int y2,
+                            int ripup_threshold,
+                            int netID,
+                            int edgeID);
+
+  bool newRipupType2(TreeEdge* treeedge,
+                            TreeNode* treenodes,
+                            int x1,
+                            int y1,
+                            int x2,
+                            int y2,
+                            int deg,
+                            int netID);
+  bool newRipup3DType3(int netID, int edgeID);
+  void newRipupNet(int netID);
+
+  // utility functions
+  void printEdge(int netID, int edgeID);
+  void ConvertToFull3DType2();
+  void fillVIA();
+  int threeDVIA();
+  void assignEdge(int netID, int edgeID, bool processDIR);
+  void recoverEdge(int netID, int edgeID);
+  void newLayerAssignmentV4();
+  void netpinOrderInc();
+  void checkRoute3D();
+  void StNetOrder();
+  bool checkRoute2DTree(int netID);
+  void checkUsage();
+  void netedgeOrderDec(int netID);
+  void printTree2D(int netID);
+  void printEdge2D(int netID, int edgeID);
+  void printEdge3D(int netID, int edgeID);
+  void printTree3D(int netID);
+  void check2DEdgesUsage();
+  void newLA();
+  void copyBR(void);
+  void copyRS(void);
+  void freeRR(void);
+  Tree fluteToTree(stt::Tree fluteTree);
+  stt::Tree treeToFlute(Tree tree);
+  int edgeShift(Tree* t, int net);
+  int edgeShiftNew(Tree* t, int net);
+
+  static const int MAXLEN = 20000;
+  static const int BIG_INT = 1e7;     // big integer used as infinity
+  static const int NOROUTE = 0;
+  static const int LROUTE = 1;
+  static const int ZROUTE = 2;
+  static const int MAZEROUTE = 3;
+  static const int HCOST =  5000;
+  static const int SAMEX =  0;
+  static const int SAMEY =  1;
+
   int maxNetDegree;
-  std::vector<int> cap_per_layer;
-  std::vector<int> usage_per_layer;
-  std::vector<int> overflow_per_layer;
-  std::vector<int> max_h_overflow;
-  std::vector<int> max_v_overflow;
+  std::vector<int> cap_per_layer_;
+  std::vector<int> usage_per_layer_;
+  std::vector<int> overflow_per_layer_;
+  std::vector<int> max_h_overflow_;
+  std::vector<int> max_v_overflow_;
   odb::dbDatabase* db_;
   bool allow_overflow_;
   int overflow_iterations_;
   int num_nets_;
   int layer_orientation_;
+  int x_range_;
+  int y_range_;
+
+  int newnetID;
```

Comment:
Why no trailing _ here , like above?

Reply:
I'll add the "_" after your reviews. I didn't want to pollute the PR with these small changes.

Reply:
That sounds good